Add the notification channels tutorial (#5464)
kyma-project/kyma: docs/monitoring/08-04-send-notifications.md

---
title: Send notifications to Slack
type: Tutorials
---

> This tutorial shows you how to configure Alertmanager to send notifications. Alertmanager supports several [notification receivers](https://prometheus.io/docs/alerting/configuration/), however, this tutorial focuses on sending notifications to Slack.

Reviewer: Maybe let's link directly to https://prometheus.io/docs/alerting/configuration/#receiver ?

Author: Yes, good idea.

> This tutorial shows you how to configure Alertmanager to send notifications. Alertmanager supports several [notification receivers](https://prometheus.io/docs/alerting/configuration/), however, this tutorial focuses on sending notifications to Slack.

Reviewer: ```suggestion
This tutorial shows you how to configure Alertmanager to send notifications. Alertmanager supports several [notification receivers](https://prometheus.io/docs/alerting/configuration/), but this tutorial only focuses on sending notifications to Slack.
```

Author: Done.

## Prerequisites

> This  tutorial is a follow-up of the [observe application metrics](/components/monitoring/#tutorials-observe-application-metrics) and the  [define alerting rules](https://kyma-project.io/docs/master/components/monitoring/#tutorials-define-alerting-rules) tutorials that use the `monitoring-custom-metrics` example. Follow this tutorial to deploy the `sample-metrics-8081` service which exposes the `cpu_temperature_celsius` metric, and creates an alert based on it. That configuration is required to complete this tutorial.

Reviewer: ```suggestion
This tutorial is a follow-up of the [**Observe application metrics**](/components/monitoring/#tutorials-observe-application-metrics) and the [**Define alerting rules**](https://kyma-project.io/docs/master/components/monitoring/#tutorials-define-alerting-rules) tutorials that use the `monitoring-custom-metrics` example. Follow this tutorial to deploy the `sample-metrics-8081` service which exposes the `cpu_temperature_celsius` metric and creates an alert based on it. That configuration is required to complete this tutorial.
```

Reviewer: (double space, tutorial name formatting changes + comma removal)

Author: Done.

## Steps

> Follow these steps to configure notifications for Slack.

Reviewer: ```suggestion
Follow these steps to configure notifications for Slack every time Alertmanager triggers and resolves the `CPUTempHigh` alert.
```

Author: Done.

> 1. Install the Incoming WebHooks Slack app and configure it to receive notifications coming from third party services. Read [this](https://api.slack.com/incoming-webhooks#create_a_webhook) document to find out how to set up the configuration. 

Reviewer: ```suggestion
1. Install the Incoming WebHooks Slack application and configure it to receive notifications coming from third-party services. Read [this](https://api.slack.com/incoming-webhooks#create_a_webhook) document to find out how to set up the configuration. 
```

Author: Done. Reworked it a bit.

  >**NOTE**: The approval of your Slack workspace administrator may be necessary to set up the webhook.

 The integration settings should look similar to the following:

 ![Integration Settings](./assets/integration-settings.png)

> 2. The configuration for notification receivers is located in [this](https://github.com/kyma-project/kyma/blob/master/resources/monitoring/charts/alertmanager/templates/alertmanager.config.yaml) template. By default, it contains settings for VictorOps, Slack, and Webhooks. Define a Secret to [override](../../../../docs/kyma/05-03-overrides.md) default [values](https://github.com/kyma-project/kyma/blob/master/resources/monitoring/charts/alertmanager/values.yaml) used by the chart.

Reviewer: ```suggestion
2. Override Alertmanager configuration.

The configuration for notification receivers is located in [this](https://github.com/kyma-project/kyma/blob/master/resources/monitoring/charts/alertmanager/templates/alertmanager.config.yaml) template. By default, it contains settings for VictorOps, Slack, and Webhooks. Define a Secret to [override](../../../../docs/kyma/05-03-overrides.md) default [values](https://github.com/kyma-project/kyma/blob/master/resources/monitoring/charts/alertmanager/values.yaml) used by the chart.
```

Author: Done + fixed link.

```yaml
apiVersion: v1
kind: Secret
metadata:
 name: monitoring-config-overrides
 namespace: kyma-installer
 labels:
    kyma-project.io/installation: ""
    installer: overrides
    component: monitoring
type: Opaque
stringData:
    global.alertTools.credentials.slack.channel: "{CHANNEL_NAME}"
    global.alertTools.credentials.slack.apiurl: "{WEBHOOK_URL}"
```
Use the following parameters:

| Parameter | Description |
|-----------|--------------------|
| **global.alertTools.credentials.slack.channel** | Specifies the Slack channel which receives notifications on new alerts, such as `test-monitoring-alerts`.

> | **global.alertTools.credentials.slack.apiurl** | Specifies the URL endpoint which sends alerts triggered by Prometheus rules. The Incoming Webhooks Slack app provides you with the Webhook URL you can paste in this configuration, such as `https://hooks.slack.com/services/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`. |

Reviewer: ```suggestion
| **global.alertTools.credentials.slack.apiurl** | Specifies the URL endpoint which sends alerts triggered by Prometheus rules. The Incoming Webhooks Slack application provides you with the Webhook URL, such as `https://hooks.slack.com/services/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`, that you can paste in this configuration. |
```

Author: Done.

---
title: Send notifications to Slack
type: Tutorials
---
This tutorial shows you how to configure Alertmanager to send notifications. Alertmanager supports several [notification receivers](https://prometheus.io/docs/alerting/configuration/#receiver), but this tutorial only focuses on sending notifications to Slack.

## Prerequisites

> This  tutorial is a follow-up of the [**observe application metrics**](/components/monitoring/#tutorials-observe-application-metrics) and the [**define alerting rules**](https://kyma-project.io/docs/master/components/monitoring/#tutorials-define-alerting-rules) tutorials that use the `monitoring-custom-metrics` example. Follow this tutorial to deploy the `sample-metrics-8081` service which exposes the `cpu_temperature_celsius` metric and creates an alert based on it. That configuration is required to complete this tutorial.

Reviewer: Let's capitalize "Observe" and "Define" as they are tutorial names.

## Steps

Follow these steps to configure notifications for Slack every time Alertmanager triggers and resolves the `CPUTempHigh` alert.

> 1. Install the Incoming WebHooks Slack application using Slack App Directory.

Reviewer: ```suggestion
1. Install the Incoming WebHooks application from Slack using Slack App Directory.
```

>   >**NOTE**: The approval of your Slack workspace administrator may be necessary install the application.

Reviewer: ```suggestion
  >**NOTE**: The approval of your Slack workspace administrator may be necessary to install the application.
```

> 2. Configure the application to receive notifications coming from third party services. Read [this](https://api.slack.com/incoming-webhooks#create_a_webhook) document to find out how to set up the configuration. 

Reviewer: ```suggestion
2. Configure the application to receive notifications coming from third-party services. Read [this](https://api.slack.com/incoming-webhooks#create_a_webhook) document to find out how to set up the configuration. 
```

2. Override Alertmanager configuration. The configuration for notification receivers is located in [this](https://github.com/kyma-project/kyma/blob/master/resources/monitoring/charts/alertmanager/templates/alertmanager.config.yaml) template. By default, it contains settings for VictorOps, Slack, and Webhooks. Define a Secret to [override](/root/kyma/#configuration-helm-overrides-for-kyma-installation) default [values](https://github.com/kyma-project/kyma/blob/master/resources/monitoring/charts/alertmanager/values.yaml) used by the chart.

```yaml
apiVersion: v1
kind: Secret
metadata:
 name: monitoring-config-overrides
 namespace: kyma-installer
 labels:
    kyma-project.io/installation: ""
    installer: overrides
    component: monitoring
type: Opaque
stringData:
    global.alertTools.credentials.slack.channel: "{CHANNEL_NAME}"
    global.alertTools.credentials.slack.apiurl: "{WEBHOOK_URL}"
```
Use the following parameters:

| Parameter | Description |
|-----------|--------------------|
| **global.alertTools.credentials.slack.channel** | Specifies the Slack channel which receives notifications on new alerts, such as `test-monitoring-alerts`.

> | **global.alertTools.credentials.slack.apiurl** | Specifies the URL endpoint which sends alerts triggered by Prometheus rules. The Incoming Webhooks Slack application provides you with the Webhook URL, such as `https://hooks.slack.com/services/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`, that you can paste in this configuration. |

Reviewer: ```suggestion
| **global.alertTools.credentials.slack.apiurl** | Specifies the URL endpoint which sends alerts triggered by Prometheus rules. The Incoming Webhooks application provides you with the Webhook URL, such as `https://hooks.slack.com/services/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`, that you can paste in this configuration. |
```

> For details on Alertmanager chart configuration and parameters see [this](components/monitoring/#configuration-alertmanager-sub-chart) document.

Reviewer: ```suggestion
For details on Alertmanager chart configuration and parameters, see [this](https://kyma-project.io/docs/master/components/monitoring/#details-alertmanager-alertmanager-configuration) document.
```

Reviewer: Full link, to master I guess.

> 4. Deploy the Secret:

Reviewer: ```suggestion
4. Deploy the Secret. Use this command:
```
